packages/builders/scripts/testing/publish-test-info.js

```diff
@@ -0,0 +1,79 @@
+import { makeTracer } from '@agoric/internal';
+import { E, Far } from '@endo/far';
+import { makeMarshal } from '@endo/marshal';
+
+const trace = makeTracer('PublishTestInfo');
+const { Fail } = assert;
+
+/**
+ * @param {BootstrapPowers} powers
+ */
+export const publishTestInfo = async powers => {
+  const {
+    consume: { agoricNamesAdmin, chainStorage: chainStorageP },
+  } = powers;
+
+  const chainStorage = await chainStorageP;
+  if (!chainStorage) {
+    trace('no chain storage, not registering chain info');
+    return;
+  }
+
+  trace('publishing testInfo');
+  const agoricNamesNode = E(chainStorage).makeChildNode('agoricNames');
+  const testInfoNode = E(agoricNamesNode).makeChildNode('testInfo');
+  const { nameAdmin } = await E(agoricNamesAdmin).provideChild('testInfo');
+
+  trace('registering onUpdate...');
+  await E(nameAdmin).onUpdate(
+    Far('chain info writer', {
+      write(entries) {
+        const marshalData = makeMarshal(_val => Fail`data only`);
+        const value = JSON.stringify(marshalData.toCapData(entries));
+        void E(testInfoNode)
+          .setValue(value)
+          .catch(() =>
+            console.log('cannot update vstorage after write to testInfo'),
+          );
+      },
+    }),
+  );
+
+  trace('writing to testInfo...');
+  await E(nameAdmin).update('agoric', {
+    isAwesome: 'yes',
+    tech: ['HardenedJs', 'Orchestration', 'Async_Execution'],
+  });
+
+  trace('Done.');
+};
+
+export const getManifestForPublishTestInfo = () => {
+  return {
+    manifest: {
+      [publishTestInfo.name]: {
+        consume: {
+          agoricNamesAdmin: true,
+          chainStorage: true,
+        },
+      },
+    },
+  };
+};
+
+/** @type {import('@agoric/deploy-script-support/src/externalTypes.js').CoreEvalBuilder} */
+export const defaultProposalBuilder = async () =>
+  harden({
+    // Somewhat unorthodox, source the exports from this builder module
```

Comment:
I don't have a problem with this.

Reply:
This is a copy from here;

https://github.com/Agoric/agoric-sdk/blob/56d1c14ed1d0cf6e9a806c21b62c90a16974c9bb/packages/builders/scripts/testing/start-valueVow.js#L72-L84

Reply:
Then it looks less unorthodox than it used to. Maybe the comment isn't necessary?